### DISCUSSION TITLE
Specify underlying S3 parameters like content-encoding

### DISCUSSION THREAD

**Discussion by mayank-kc** [BetaHuhn/do-spaces-action#237] [Ideas]
It will be great to provide meta-data options like content-encoding

**Comment by BetaHuhn** [comment]
Hey, thanks for taking the time to open this discussion!

> It will be great to provide meta-data options like content-encoding

What use case do you have for the `content-encoding`?

We could add way to expose all of the available options of the [s3 PutObject request](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html#API_PutObject_RequestParameters). 

Maybe a `s3_options` key like this:

```yaml
with:
   src: 'path/to/file.txt'
   s3_options:
      content-encoding: 'gzip'
```

which would be passed directly to the underlying aws sdk: https://github.com/BetaHuhn/do-spaces-action/blob/af5a1222fcb2ef81d3c05ce060b86b2b87c2bf09/src/interface.js#L25-L33

Would this work for you?

I have very limited time right now to implement this, @mayank-kc would you be willing to create a PR for this?
